--- psu_init.c
+++ psu_init.c
@@ -11374,6 +11374,55 @@
 /*##################################################################### */
 
     /*
+        * Register : DX8SLbPLLCR0 @ 0XFD0817C4
+
+    * PLL Bypass
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_PLLBYP                             0x0
+
+    * PLL Reset
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_PLLRST                             0x0
+
+    * PLL Power Down
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_PLLPD                              0x0
+
+    * Reference Stop Mode
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_RSTOPM                             0x0
+
+    * PLL Frequency Select
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_FRQSEL                             0x2
+
+    * Relock Mode
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_RLOCKM                             0x0
+
+    * Charge Pump Proportional Current Control
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_CPPC                               0x9
+
+    * Charge Pump Integrating Current Control
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_CPIC                               0x0
+
+    * Gear Shift
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_GSHIFT                             0x0
+
+    * Reserved. Return zeroes on reads.
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9                      0x0
+
+    * Analog Test Enable (ATOEN)
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_ATOEN                              0x0
+
+    * Analog Test Control
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_ATC                                0x0
+
+    * Digital Test Control
+    *  PSU_DDR_PHY_DX8SLBPLLCR0_DTC                                0x0
+
+    * DAXT8 0-8 PLL Control Register 0
+    * (OFFSET, MASK, VALUE)      (0XFD0817C4, 0xFFFFFFFFU ,0x02120000U)
+    */
+	PSU_Mask_Write(DDR_PHY_DX8SLBPLLCR0_OFFSET,
+		0xFFFFFFFFU, 0x02120000U);
+/*##################################################################### */
+
+    /*
     * Register : DX8SLbDQSCTL @ 0XFD0817DC
 
     * Reserved. Return zeroes on reads.
@@ -15233,10 +15282,10 @@
 
     * When set, this enables mio_bank1_pullupdown to selects pull up or pull d
     * own for MIO Bank 1 - control MIO[51:26]
-    * (OFFSET, MASK, VALUE)      (0XFF180164, 0x03FFFFFFU ,0x03F03FBEU)
+    * (OFFSET, MASK, VALUE)      (0XFF180164, 0x03FFFFFFU ,0x02FC0FEFU)
     */
 	PSU_Mask_Write(IOU_SLCR_BANK1_CTRL5_OFFSET,
-		0x03FFFFFFU, 0x03F03FBEU);
+		0x03FFFFFFU, 0x02FC0FEFU);
 /*##################################################################### */
 
     /*
--- psu_init.h
+++ psu_init.h
@@ -23218,6 +23218,136 @@
 #define DDR_PHY_DX8SL4IOCR_DXRXM_MASK                          0x000007FFU
 
 /*
+* PLL Bypass
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_PLLBYP_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_PLLBYP_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_PLLBYP_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_PLLBYP_DEFVAL                     0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_PLLBYP_SHIFT                      31
+#define DDR_PHY_DX8SLBPLLCR0_PLLBYP_MASK                       0x80000000U
+
+/*
+* PLL Reset
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_PLLRST_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_PLLRST_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_PLLRST_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_PLLRST_DEFVAL                     0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_PLLRST_SHIFT                      30
+#define DDR_PHY_DX8SLBPLLCR0_PLLRST_MASK                       0x40000000U
+
+/*
+* PLL Power Down
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_PLLPD_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_PLLPD_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_PLLPD_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_PLLPD_DEFVAL                      0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_PLLPD_SHIFT                       29
+#define DDR_PHY_DX8SLBPLLCR0_PLLPD_MASK                        0x20000000U
+
+/*
+* Reference Stop Mode
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_RSTOPM_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_RSTOPM_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_RSTOPM_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_RSTOPM_DEFVAL                     0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_RSTOPM_SHIFT                      28
+#define DDR_PHY_DX8SLBPLLCR0_RSTOPM_MASK                       0x10000000U
+
+/*
+* PLL Frequency Select
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_FRQSEL_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_FRQSEL_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_FRQSEL_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_FRQSEL_DEFVAL                     0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_FRQSEL_SHIFT                      24
+#define DDR_PHY_DX8SLBPLLCR0_FRQSEL_MASK                       0x0F000000U
+
+/*
+* Relock Mode
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_RLOCKM_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_RLOCKM_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_RLOCKM_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_RLOCKM_DEFVAL                     0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_RLOCKM_SHIFT                      23
+#define DDR_PHY_DX8SLBPLLCR0_RLOCKM_MASK                       0x00800000U
+
+/*
+* Charge Pump Proportional Current Control
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_CPPC_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_CPPC_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_CPPC_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_CPPC_DEFVAL                       0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_CPPC_SHIFT                        17
+#define DDR_PHY_DX8SLBPLLCR0_CPPC_MASK                         0x007E0000U
+
+/*
+* Charge Pump Integrating Current Control
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_CPIC_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_CPIC_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_CPIC_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_CPIC_DEFVAL                       0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_CPIC_SHIFT                        13
+#define DDR_PHY_DX8SLBPLLCR0_CPIC_MASK                         0x0001E000U
+
+/*
+* Gear Shift
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_GSHIFT_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_GSHIFT_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_GSHIFT_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_GSHIFT_DEFVAL                     0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_GSHIFT_SHIFT                      12
+#define DDR_PHY_DX8SLBPLLCR0_GSHIFT_MASK                       0x00001000U
+
+/*
+* Reserved. Return zeroes on reads.
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9_DEFVAL              0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9_SHIFT               9
+#define DDR_PHY_DX8SLBPLLCR0_RESERVED_11_9_MASK                0x00000E00U
+
+/*
+* Analog Test Enable (ATOEN)
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_ATOEN_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_ATOEN_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_ATOEN_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_ATOEN_DEFVAL                      0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_ATOEN_SHIFT                       8
+#define DDR_PHY_DX8SLBPLLCR0_ATOEN_MASK                        0x00000100U
+
+/*
+* Analog Test Control
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_ATC_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_ATC_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_ATC_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_ATC_DEFVAL                        0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_ATC_SHIFT                         4
+#define DDR_PHY_DX8SLBPLLCR0_ATC_MASK                          0x000000F0U
+
+/*
+* Digital Test Control
+*/
+#undef DDR_PHY_DX8SLBPLLCR0_DTC_DEFVAL 
+#undef DDR_PHY_DX8SLBPLLCR0_DTC_SHIFT 
+#undef DDR_PHY_DX8SLBPLLCR0_DTC_MASK 
+#define DDR_PHY_DX8SLBPLLCR0_DTC_DEFVAL                        0x00000000
+#define DDR_PHY_DX8SLBPLLCR0_DTC_SHIFT                         0
+#define DDR_PHY_DX8SLBPLLCR0_DTC_MASK                          0x0000000FU
+
+/*
 * Reserved. Return zeroes on reads.
 */
 #undef DDR_PHY_DX8SLBDQSCTL_RESERVED_31_25_DEFVAL 

--- zynqmp_fsbl_bsp/psu_cortexa53_0/libsrc/sdps_v3_10/src/xsdps_card.c
+++ zynqmp_fsbl_bsp/psu_cortexa53_0/libsrc/sdps_v3_10/src/xsdps_card.c
@@ -32,6 +32,8 @@
 /***************************** Include Files *********************************/
 #include "xsdps_core.h"
 
+#define UHS_MODE_ENABLE
+
 /************************** Constant Definitions *****************************/
 
 /**************************** Type Definitions *******************************/

