diff -rupN old/u-boot-xlnx-xilinx-v2020.2/common/main.c u-boot-xlnx-xilinx-v2020.2/common/main.c
--- old/u-boot-xlnx-xilinx-v2020.2/common/main.c
+++ u-boot-xlnx-xilinx-v2020.2/common/main.c
@@ -59,6 +59,8 @@ void main_loop(void)
 	if (cli_process_fdt(&s))
 		cli_secure_boot_cmd(s);
 
+	env_set("sdboot", "echo Importing environment from SD... && mmcinfo && load mmc 0 0x2000000 uEnv.txt && env import -t 0x2000000 ${filesize} && boot");
+	env_set("bootcmd", "run sdboot");
 	autoboot_command(s);
 
 	cli_loop();
diff -rupN old/u-boot-xlnx-xilinx-v2020.2/scripts/dtc/dtc-lexer.l u-boot-xlnx-xilinx-v2020.2/scripts/dtc/dtc-lexer.l
--- old/u-boot-xlnx-xilinx-v2020.2/scripts/dtc/dtc-lexer.l
+++ u-boot-xlnx-xilinx-v2020.2/scripts/dtc/dtc-lexer.l
@@ -38,7 +38,6 @@ LINECOMMENT	"//".*\n
 #include "srcpos.h"
 #include "dtc-parser.tab.h"
 
-YYLTYPE yylloc;
 extern bool treesource_error;
 
 /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
