--- src/runtime_src/core/edge/drm/zocl/Makefile
+++ src/runtime_src/core/edge/drm/zocl/Makefile
@@ -80,6 +80,9 @@
 
 all: modules
 
+ccflags-y := -I$(src)/include -I$(src)/../../include -I$(src)/../../../include -I$(src)/../../../common/drv/include -DXRT_HASH="\"$(GIT_HASH)\"" -DXRT_HASH_DATE="\"$(GIT_HASH_DATE)\"" -DXRT_BRANCH="\"$(GIT_BRANCH)\"" -DXRT_MODIFIED_FILES="\"$(GIT_MODIFIED_FILES)\"" -DXRT_DATE="\"$(DATE_NOW)\"" -DXRT_DRIVER_VERSION="\"$(XRT_DRIVER_VERSION)\""
+
+
 git_variables:
 	$(eval GIT_HASH ?= $(shell git rev-parse --verify HEAD))
 	$(eval GIT_HASH_DATE ?= $(shell git log -1 --pretty=format:%cD))

